src/Search.php

```diff
@@ -5461,6 +5461,9 @@ public static function addWhere($link, $nott, $itemtype, $ID, $searchtype, $val,
                 $addcondition = $addcondition . " ";
             }
 
+            if ($beforejoin['table'] === $link_table && isset($beforejoin['joinparams']['field'])) {
+                $fk = $beforejoin['joinparams']['field'];
+            }
```

Comment:
Please add an explanation in the comments, we may be asking questions in the future.